### PR TITLE
Category page layout

### PR DESCRIPTION
**Pull request by siggakogk**
# Category page updates

Using new sidebarlayout
Using new navigation component
Using new routing 

## Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/web/screens/Category/Category.tsx [hunk 1]
```diff
@@ -268,184 +253,208 @@ const Category: Screen<CategoryProps> = ({
...
-        sidebar={
-          <Sidebar
-            bullet="none"
+      <SidebarLayout
+        isSticky
+        sidebarContent={
+          <Navigation
+            baseId="desktopNav"
+            colorScheme="purple"
+            isMenuDialog={false}
```
Comment (barabrian) [inline apps/web/screens/Category/Category.tsx:262]: This is default false so probably no need to define it here

File: apps/web/screens/Category/Category.tsx [hunk 2]
```diff
@@ -268,184 +253,208 @@ const Category: Screen<CategoryProps> = ({
...
-        </Text>
-        <Text variant="intro" as="p">
-          {category.description}
-        </Text>
-      </CategoryLayout>
+        <Box display={['block', 'block', 'none']} marginBottom={4}>
+          <Navigation
+            baseId="mobileNav"
+            colorScheme="purple"
+            isMenuDialog={true}
```
Comment (barabrian) [inline apps/web/screens/Category/Category.tsx:297]: no need adding `={true}` for booleans
```suggestion
            isMenuDialog
```

File: apps/web/screens/Category/Category.tsx [hunk 3]
```diff
@@ -268,184 +253,208 @@ const Category: Screen<CategoryProps> = ({
...
+                  {...pathNames(activeLocale, typename as ContentType, slug)}
+                  passHref
+                >
+                  {link}
+                </NextLink>
+              )
+            }}
+            items={sidebarCategoryLinks}
+            title={n('sidebarHeader')}
+            activeItemTitle={n('sidebarHeader')}
```
Comment (barabrian) [inline apps/web/screens/Category/Category.tsx:310]: Shouldn't this be current active title? to be displayed in the dialog version?

File: apps/web/screens/Category/Category.tsx [hunk 4]
```diff
@@ -139,17 +124,14 @@ const Category: Screen<CategoryProps> = ({
...
-    value: c.slug,
-  }))
+  const sidebarCategoryLinks = categories.map(({ __typename, title, slug }) => {
+    return {
+      title,
+      typename: __typename,
+      active: slug === Router.query.slug,
+      slug: [slug],
+    }
+  })
```
Comment (barabrian) [inline apps/web/screens/Category/Category.tsx:134]: I'm not sure how the category pages work, but in Article pages we need the parent article for the first link, is that maybe not the case here? or is this maybe a different setup?

**Review by barabrian — COMMENTED** [review]
Minor comments, otherwise it looks good to me

File: apps/web/screens/Category/Category.tsx (hunk 4 above)
Comment (siggakogk) [reply apps/web/screens/Category/Category.tsx:134]: Yes, but it's not the case here. In the navbar here we display all categories, no subcategories or something of that sort.

File: apps/web/screens/Category/Category.tsx (hunk 3 above)
Comment (siggakogk) [reply apps/web/screens/Category/Category.tsx:310]: Ahh yes, my bad!

File: apps/web/screens/Category/Category.tsx [hunk 5]
```diff
@@ -268,184 +253,208 @@ const Category: Screen<CategoryProps> = ({
...
+              return (
+                <NextLink
+                  {...pathNames(activeLocale, typename as ContentType, slug)}
+                  passHref
+                >
+                  {link}
+                </NextLink>
+              )
+            }}
+            activeItemTitle={n('sidebarHeader')}
```
[the remark this answers is not in the document]
Comment (siggakogk) [reply apps/web/screens/Category/Category.tsx:275]: Yep not needed, will remove!

**Review by barabrian — APPROVED** [review]
LGTM

**Pull request merged**